exec-unittests

```diff
@@ -186,18 +181,30 @@ for _major in ${major_list}; do
                 (( count++ ))
             fi
         done > ${_para_jobs_file}
-        parallel --will-cite -k --lb -a ${_para_jobs_file} ${para_jobs_arg} para_run_legacy
-        if [[ ${?} -ne 0 ]]; then
-            rc=1
-        fi
-        if [[ ${count} -eq 0 && "${subtst}" != "python" ]]; then
-            printf -- "Error - unrecognized sub-test, '%s'\n" "${subtst}" >&2
-            rc=1
+        if [[ -s ${_para_jobs_file} ]]; then
+			parallel --will-cite -k --lb -a ${_para_jobs_file} ${para_jobs_arg} para_run_legacy
+			if [[ ${?} -ne 0 ]]; then
+				rc=1
+			fi
+			if [[ ${count} -eq 0 && "${subtst}" != "python" ]]; then
+				printf -- "Error - unrecognized sub-test, '%s'\n" "${subtst}" >&2
+				rc=1
+			fi
         fi
+
+		# Verify the Agent Makefile source build functions correctly.
+		if [[ "${subtst:-srcbld}" == "srcbld" ]]; then
+			verify_make_source_tree ${_major} || rc=1
+		fi
     elif [[ "${_major}" == "server" ]]; then
         if [[ "${subtst}" == "legacy" ]]; then
             printf -- "Warning - server legacy tests are no longer executed\n" >&2
         fi
+
+		# Verify the Server Makefile source build functions correctly.
+		if [[ "${subtst:-srcbld}" == "srcbld" ]]; then
+			verify_make_source_tree ${_major} || rc=1
+		fi
```

Comment:
Do we still want to do this, now that the CI is building the RPMs for the Agent and the Server, whose build process relies on the source-build?

Reply:
I like the immediate feedback that the source build breakage happens early in the pipeline and more accessible to the developer who doesn't have to run much to get the feedback.

If folks would rather this in one place, I'd be happy to rip it out.

Reply:
No -- I can see the value of this -- we probably shouldn't rip it out entirely.  However, I'm not convinced that the immediacy is worth imposing the cost in all cases.

I noticed in my CI work that this step takes a non-trivial amount of time.  So, if (for example) we are going to proceed on to running the RPM build anyway, it's not helpful to perform this step here.

By the way, I noticed a couple of things.  First, we're performing this step even if the preceding step failed.  In the name of immediacy, it seems like we should be proceeding to report that failure instead of continuing with this step.  Second, I _think_ that the very next thing the CI does after this step is to build the RPMs, so doing this here (in that case) doesn't actually speed the reporting of the failure, it just delays the completion of the build.

So, I have two suggestions:  first, remove the defaulting -- if `srcbld` is requested, then run the step; second, add to the conditional so that this step is skipped if `rc` is non-zero.

Reply:
I routine type `tox -e py39 -- agent` to verify all the agent code works.  I want to be sure that I don't break the source tree builds each time I do that.  If we remove `srcbld` as the default, I'll have to wait for the CI job failure to catch a problem.

I'd rather have a simple `cd agent/rpm; make rpm` verify it is all working, but that is out of the scope of this PR.

So as I think about this, I'd rather keep this as is and not remove the default.